apps/gyrokinetic.c

```diff
@@ -873,27 +878,90 @@ gkyl_gyrokinetic_app_write_rad_emissivity(gkyl_gyrokinetic_app* app, int sidx, d
-    char fileNm_emissivity[sz_emissivity+1]; // ensures no buffer overflow
-    snprintf(fileNm_emissivity, sizeof fileNm_emissivity, fmt_emissivity, app->name,
-      s->info.name, app->species[s->rad.collide_with_idx[i]].info.name, frame);
-    gkyl_comm_array_write(s->comm, &app->grid, &app->local, mt, s->rad.emissivity_host[i], fileNm_emissivity);
+    const char *fmt_emissivity = "%s-%s_emissivity_%s_%d.gkyl";  
+    if (s->rad.is_neut_species[i]) {
+      int sz_emissivity = gkyl_calc_strlen(fmt_emissivity, app->name, s->info.name, app->neut_species[s->rad.collide_with_idx[i]].info.name, frame);
+      char fileNm_emissivity[sz_emissivity+1]; // ensures no buffer overflow
+      snprintf(fileNm_emissivity, sizeof fileNm_emissivity, fmt_emissivity, app->name, s->info.name, app->neut_species[s->rad.collide_with_idx[i]].info.name, frame);
+      gkyl_comm_array_write(app->comm, &app->grid, &app->local, mt, s->rad.emissivity_host[i], fileNm_emissivity);
+    } else {
+      int sz_emissivity = gkyl_calc_strlen(fmt_emissivity, app->name, s->info.name, app->species[s->rad.collide_with_idx[i]].info.name, frame);
+      char fileNm_emissivity[sz_emissivity+1]; // ensures no buffer overflow
+      snprintf(fileNm_emissivity, sizeof fileNm_emissivity, fmt_emissivity, app->name, s->info.name, app->species[s->rad.collide_with_idx[i]].info.name, frame);
+      gkyl_comm_array_write(app->comm, &app->grid, &app->local, mt, s->rad.emissivity_host[i], fileNm_emissivity);
+    }  
   }
 
   gyrokinetic_array_meta_release(mt);   
 }
 
+void
+gkyl_gyrokinetic_app_write_rad_integrated_moms(gkyl_gyrokinetic_app *app, int sidx, double tm)
+{
+  int vdim = app->vdim;
+  double avals_global[2+vdim];
+  struct timespec wst = gkyl_wall_clock();
+
+  struct gk_species *gk_s = &app->species[sidx];
+  // Compute radiation drag coefficients
+  const struct gkyl_array *fin_neut[app->num_neut_species];
+  const struct gkyl_array *fin[app->num_species];
+  for (int i=0; i<app->num_species; ++i) 
+    fin[i] = app->species[i].f;
+  for (int i=0; i<app->num_neut_species; ++i)
+    fin_neut[i] = app->neut_species[i].f;
+  gk_species_radiation_moms(app, gk_s, &gk_s->rad, fin, fin_neut);
+  gk_species_radiation_integrated_moms(app, gk_s, &gk_s->rad, fin, fin_neut);
+
+  // reduce to compute sum over whole domain, append to diagnostics
+  gkyl_array_reduce_range(gk_s->rad.red_integ_diag, gk_s->rad.integ_moms.marr, GKYL_SUM, &app->local);
+  gkyl_comm_allreduce(app->comm, GKYL_DOUBLE, GKYL_SUM, 2+vdim, 
+    gk_s->rad.red_integ_diag, gk_s->rad.red_integ_diag_global);
+  if (app->use_gpu) {
+    gkyl_cu_memcpy(avals_global, gk_s->rad.red_integ_diag_global, sizeof(double[2+vdim]), GKYL_CU_MEMCPY_D2H);
+  }
+  else {
+    memcpy(avals_global, gk_s->rad.red_integ_diag_global, sizeof(double[2+vdim]));
+  }
+  gkyl_dynvec_append(gk_s->rad.integ_diag, tm, avals_global);
+
+  // Write from rank 0
+  int rank;
```

Comment:
@jRoeltgen @akashukla totally sure, but I think this should be `rank[1]`.

Reply:
So long as you pass rank by reference to mpi_comm_rank, it's fine. You do not need to make this explicitly rank[1] anywhere (and we do not do this in any of the other dynvec write methods). 

The only reason to make it rank[1] is if you want to use the implicit pass by reference in this case (which I don't think we should be doing anyway).